tests/main/nvidia-files/task.yaml

```diff
@@ -0,0 +1,239 @@
+        echo "Broken driver is in use, expecting: skip.d/$SPREAD_SYSTEM/$PACKAGE_VERSION${PACKAGE_SUFFIX:-} to say: broken-driver"
+        test "$(cat "skip.d/$SPREAD_SYSTEM/$PACKAGE_VERSION${PACKAGE_SUFFIX:-}")" = "broken-driver"
+        exit 0
+        ;;
+    esac
+
+    # At this step, we we expect this test to work, and no skip file to exist.
+    echo "Everything is good, expecting skip.d/$SPREAD_SYSTEM/$PACKAGE_VERSION${PACKAGE_SUFFIX:-} not to exist"
+    test ! -e "skip.d/$SPREAD_SYSTEM/$PACKAGE_VERSION${PACKAGE_SUFFIX:-}"
+
+    # We will need to install i386 libraries. This is specifically done on an
+    # amd64 system as there are cases of 32bit programs running through
+    # otherwise 64bit snap, running on 64bit host.
+    dpkg --add-architecture i386
+    apt-get update
+
+    # Install Nvidia userspace libraries at the designated version.
+    apt-get install -y \
+        libnvidia-common-"$PACKAGE_VERSION${PACKAGE_SUFFIX:-}" \
+        libnvidia-compute-"$PACKAGE_VERSION${PACKAGE_SUFFIX:-}":amd64 \
+        libnvidia-compute-"$PACKAGE_VERSION${PACKAGE_SUFFIX:-}":i386 \
+        libnvidia-decode-"$PACKAGE_VERSION${PACKAGE_SUFFIX:-}":amd64 \
+        libnvidia-decode-"$PACKAGE_VERSION${PACKAGE_SUFFIX:-}":i386 \
+        libnvidia-encode-"$PACKAGE_VERSION${PACKAGE_SUFFIX:-}":amd64 \
+        libnvidia-encode-"$PACKAGE_VERSION${PACKAGE_SUFFIX:-}":i386 \
+        libnvidia-fbc1-"$PACKAGE_VERSION${PACKAGE_SUFFIX:-}":amd64 \
+        libnvidia-fbc1-"$PACKAGE_VERSION${PACKAGE_SUFFIX:-}":i386 \
+        libnvidia-gl-"$PACKAGE_VERSION${PACKAGE_SUFFIX:-}":amd64 \
+        libnvidia-gl-"$PACKAGE_VERSION${PACKAGE_SUFFIX:-}":i386
+
+    # Look at the canary file libnvidia-glcore.so.* to get the exact version of
+    # the driver. This file is also used by snap-confine, as a pre-condition
+    # that the libraries are installed.
+    DRIVER_VERSION="$(find /usr/lib/x86_64-linux-gnu/ -name 'libnvidia-glcore.so.*' | sed -e 's,.*/libnvidia-glcore\.so\.,,')"
+
+    # Pretend we have Nvidia kernel module loaded, so that snap-confine enables
+    # special logic. The actual version we pretend to have is set later, as it
+    # must match installed libraries so that the right canary file is detected by
+    # snap-confine.
+    mkdir -p /tmp/sys-module/nvidia
+    echo "$DRIVER_VERSION" >/tmp/sys-module/nvidia/version
+    if os.query is-xenial || os.query is-bionic; then
+        mount -o bind /tmp/sys-module/ /sys/module
+    else
+        mkdir -p /tmp/sys-module-work
+        LO="/sys/module/"
+        UP="/tmp/sys-module"
+        WORK="/tmp/sys-module-work"
+        mount -t overlay overlay \
+            -olowerdir="$LO",upperdir="$UP",workdir="$WORK" /sys/module
+    fi
+    tests.cleanup defer rm -rf /tmp/sys-module
+    tests.cleanup defer umount /sys/module
+
+    if [ -f test-snapd-nvidia_0.1.0_amd64.snap ]; then
```

Comment:
perhaps we could remove this if too, as snaps cannot be part of the spread content

Reply:
fixed